### PR TITLE
Migration to Android Embedding v2

### LINKED ISSUE #60: Any upgrade?
Please, continue...

### PR DESCRIPTION
**Pull request by rwrz**
- changed the way the plugin handles the app lifecycle to a kotlin Channel
- implemented new plugin methods
- kept old registar for old app as well (and tested)
- updated example to use v2
- removal of some generated files from git

Fixes #60 and #57

### CODE REVIEW

**Comment by rwrz** [comment]
@hoanglm4 as promised, here is the new version. I also upgrade the Kotlin version, Gradle and tested (only) on the Beta Flutter repository.
Let me know if need any change, since I changed the way it listens to the app lifecycle, removing the dependency with the "registar".

**Comment by pastordee** [comment]
> @hoanglm4 as promised, here is the new version. I also upgrade the Kotlin version, Gradle and tested (only) on the Beta Flutter repository.
> Let me know if need any change, since I changed the way it listens to the app lifecycle, removing the dependency with the "registar".

Please when is this going to go live?

File: android/src/main/kotlin/com/hoanglm/flutteryoutubeview/FlutterYoutubeView.kt [hunk 1]
```diff
@@ -16,42 +13,43 @@ import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.PlayerConstan
...
+import kotlinx.coroutines.launch
 
 class FlutterYoutubeView(
         private val context: Context,
         id: Int,
         private val params: HashMap<String, *>,
-        private val state: AtomicReference<Lifecycle.Event>,
-        private val registrar: PluginRegistry.Registrar
+        private val binaryMessenger: BinaryMessenger,
+        private val lifecycleChannel: Channel<Lifecycle.Event>
```
Comment (hoanglm4) [inline android/src/main/kotlin/com/hoanglm/flutteryoutubeview/FlutterYoutubeView.kt:30]: Could you using StateFlow? Channel don't deliver last value, so lastLifecycle value is not correctly

File: android/src/main/kotlin/com/hoanglm/flutteryoutubeview/FlutterYoutubeViewPlugin.kt [hunk 2]
```diff
@@ -3,79 +3,142 @@ package com.hoanglm.flutteryoutubeview
...
+        binding.platformViewRegistry.registerViewFactory(
+                "plugins.hoanglm.com/youtube",
+                YoutubeFactory(binding.binaryMessenger, lifecycleChannel)
+        )
+    }
+
+    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+        // do nothing
+        activityBinding?.activity?.application?.unregisterActivityLifecycleCallbacks(this)
+    }
```
Comment (hoanglm4) [inline android/src/main/kotlin/com/hoanglm/flutteryoutubeview/FlutterYoutubeViewPlugin.kt:33]: activityBinding is null (onDetachedFromActivity() is called before), so unregisterActivityLifecycleCallbacks() is never called.

Could you call it on onDetachedFromActivity() ?

File: android/src/main/kotlin/com/hoanglm/flutteryoutubeview/FlutterYoutubeViewPlugin.kt [hunk 3]
```diff
@@ -3,79 +3,142 @@ package com.hoanglm.flutteryoutubeview
...
+    // This call will be followed by onReattachedToActivityForConfigChanges().
+    override fun onDetachedFromActivity() {
+        activityBinding = null
+    }
+
+    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
+        activityBinding = binding
+        registrarActivityHashCode = binding.activity.hashCode()
+        binding.activity.application.registerActivityLifecycleCallbacks(this)
+    }
```
Comment (hoanglm4) [inline android/src/main/kotlin/com/hoanglm/flutteryoutubeview/FlutterYoutubeViewPlugin.kt:44]: `activityBinding = binding
        registrarActivityHashCode = binding.activity.hashCode()
        binding.activity.application.registerActivityLifecycleCallbacks(this)`
Could you push them to onAttachedToActivity() ? on onReattachedToActivityForConfigChanges() method will call onAttachedToActivity()

**Review by hoanglm4 — COMMENTED** [review]
@rwrz see all good, but I have small comment. Could you help me check

**Pull request merged**